nnvm/src/compiler/graph_fuse.cc

```diff
@@ -161,6 +161,44 @@ nnvm::Graph GraphFusePartition(nnvm::Graph g) {
       }
     }
   }
+
+  if (opt_level >= 1) {
+    std::vector<std::vector<uint32_t>> children_group_ids(idx.num_nodes());
```

Comment:
we need to document more on the algorithm here. So far I am not certain about the entire logic

Reply:
I explained my logic in the description of this PR, is there something not clear?
But I agree that code alone is not enough to understand what I am trying to do here.

Reply:
OK, can you also add comment block about the logic being used in here?

Reply:
sure, will do.